tensorflow/core/kernels/segment_reduction_ops.h

```diff
@@ -16,6 +16,14 @@ limitations under the License.
 #ifndef THIRD_PARTY_TENSORFLOW_CORE_KERNELS_SEGMENT_REDUCTION_OPS_H_
 #define THIRD_PARTY_TENSORFLOW_CORE_KERNELS_SEGMENT_REDUCTION_OPS_H_
 
+
+// This file requires the following include because it uses CudaAtomicMax:
+// #include "tensorflow/core/util/cuda_kernel_helper.h"
```

Comment:
What if we guard the include with "ifdef GOOGLE_CUDA" ?

Reply:
I tried doing that, but discovered that CPU builds also set GOOGLE_CUDA (see [tensorflow.bzl#1](https://github.com/tensorflow/tensorflow/blob/9054c9b2ac303cbd1538166d0821f389cbc75894/tensorflow/tensorflow.bzl#L913) and [tensorflow.bzl#2](https://github.com/tensorflow/tensorflow/blob/9054c9b2ac303cbd1538166d0821f389cbc75894/tensorflow/tensorflow.bzl#L203), both seem to propagate to '.cc' files as well as '.cu.cc' files)
Probably `tensorflow.bzl` shouldn't set `GOOGLE_CUDA`, as this define is handled by the [GPU crosstool] (https://github.com/tensorflow/tensorflow/blob/9054c9b2ac303cbd1538166d0821f389cbc75894/third_party/gpus/cuda/build_defs.bzl.tpl#L18).

However when I tried removing '-DGOOGLE_CUDA=1' from `tensorflow.bzl`, but I got a bunch of link errors and didn't investigate any further. Even if we want to dig deeper into this, it would be nice to have a workaround to unbreak cuda_clang while we investigate.

Reply:
You are right, that seems like a bug.
If this works as is now, I am OK with merging it.